lightgbm/src/test/scala/com/microsoft/azure/synapse/ml/lightgbm/split1/VerifyLightGBMClassifier.scala

```diff
@@ -641,6 +641,31 @@ class VerifyLightGBMClassifier extends Benchmarks with EstimatorFuzzing[LightGBM
     assertFitWithoutErrors(baseModel, df)
   }
 
+  test("Verify LightGBM Classifier won't get stuck on " +
+    "number of features in data is not the same as it was in training data") {
+    val inputData = Seq(
+      LabeledPoint(1.0, Vectors.dense(1.0, 0.0, 3.0)),
+      LabeledPoint(0.0, Vectors.sparse(3, Array(0, 2), Array(1.0, 3.0)))
+    )
+    val testData = Seq(
+      (
+        "uuid-value",
+        Vectors.sparse(15, Array(2, 6, 8, 14), Array(1.0, 1.0, 1.0, 2.0))
+      )
+    )
+
+    val modelDF = spark.createDataFrame(inputData).toDF("labels", "features")
+    val testDF = spark.createDataFrame(testData).toDF("uuid", "features")
+
+    val fitModel = baseModel.fit(modelDF)
+    val oldModelString = fitModel.getModel.modelStr.get
+
+    val testModel = LightGBMClassificationModel.loadNativeModelFromString(oldModelString)
+    testModel.setPredictDisableShapeCheck(true)
+
+    assert(testModel.transform(testDF).collect().length > 0)
```

Comment:
nice test!

Reply:
Hmm, have problem with python tests.
I found that repo have Dockerfiles for developer.
I'm continue investigate with docker env about this.
Scala tests seems ok.

Reply:
Ya, look like all things done.

Thank you for your support in my first PR here.